ext/reflection/php_reflection.c

```diff
@@ -4447,7 +4447,7 @@ ZEND_METHOD(ReflectionClass, getConstants)
 	ZEND_HASH_FOREACH_STR_KEY_PTR(&ce->constants_table, key, constant) {
 		if (UNEXPECTED(zval_update_constant_ex(&constant->value, ce) != SUCCESS)) {
 			zend_array_destroy(Z_ARRVAL_P(return_value));
```

Comment:
You also need to drop the zend_array_destroy, otherwise there will be a double free.